### PR TITLE
Add configuration for splitting upgrade jobs instead of using Jenkins steps

### PR DESCRIPTION
**Pull request by ikehz**
Let me know what you think.

### CODE REVIEW

**Comment by ihmccreery** [comment]
Toward #22104.

File: hack/jenkins/e2e-runner.sh [hunk 1]
```diff
@@ -208,21 +224,28 @@ if [[ "${E2E_UP,,}" == "true" ]]; then
...
+        E2E_REPORT_PREFIX='upgrade' e2e_test "${GINKGO_UPGRADE_TEST_ARGS}"
+	# If JENKINS_USE_OLD_TESTS is set, back out into the old tests now that we've upgraded.
+        if [[ "${JENKINS_USE_OLD_TESTS:-}" == "true" ]]; then
+            cd ../kubernetes_old
+        fi
     fi
 fi
 
+if [[ "${E2E_TEST,,}" == "true" ]]; then
+    e2e_test "${GINKGO_TEST_ARGS:-ci/latest}"
```
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: I'm confused about why we're putting in `ci/latest` as test args.

**Comment by ihmccreery** [comment]
@spxtr Fixed; PTAL.

File: hack/jenkins/e2e-runner.sh [hunk 2]
```diff
@@ -208,21 +224,28 @@ if [[ "${E2E_UP,,}" == "true" ]]; then
...
-        echo "Publish build_version to ci/latest-green.txt: ${build_version}"
-        gsutil cp ./version gs://kubernetes-release/ci/latest-green.txt
+# Allow download & unpack of alternate version of tests, for cross-version & upgrade testing.
+if [[ -n "${JENKINS_PUBLISHED_CROSS_VERSION:-}" ]]; then
+    cd ..
+    mv kubernetes kubernetes_old
+    fetch_published_version_tars "${JENKINS_PUBLISHED_CROSS_VERSION}"
+    cd kubernetes
+    # Upgrade the cluster before running other tests
+    if [[ -n "${GINKGO_UPGRADE_TEST_ARGS:-}" ]]; then
```
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: I'm not a huge fan of using the emptiness of `GINKGO_UPGRADE_TEST_ARGS` to switch on/off running the extra `e2e_test` step. The alternative is using a separate `E2E_UPGRADE_TEST` that we explicitly set in `post-env` for all jobs. I'm not totally sure that that's better, though.

Comment (ihmccreery) [inline hack/jenkins/e2e-runner.sh]: Yeah, it makes me feel a bit funny too, but I also don't know if `E2E_UPGRADE_TEST` is better.  Happy to add it if you think it is.

Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: It's probably fine as-is.

We're actually pretty close to being able to `source e2e-runner.sh && fetch_published_version_tars && unpack_tars && (up) && e2e_test && (down)`.

**Comment by ihmccreery** [comment]
@spxtr PTAL.

**Comment by spxtr** [comment]
LGTM. Probably manually merge and make sure we don't bork Jenkins.

**Comment by ihmccreery** [comment]
Yes, let's wait until after tomorrow morning.

**Comment by ihmccreery** [comment]
@spxtr PTAL; had to rebase, and also realized that `[Feature:ClusterUpgrade]` wasn't implemented.

**Comment by spxtr** [comment]
@k8s-bot test this issue: #21990

**Comment by ihmccreery** [comment]
This is the beginnings of #22104.

**Comment by ihmccreery** [comment]
@k8s-bot test this again please: issue #23879

**Comment by spxtr** [comment]
This looks good to me. It's hard to know for sure since very little of it gets hit by CI, so maybe there's some bash variable that doesn't line up or something, but it shouldn't break the world.

**Comment by ihmccreery** [comment]
Merging manually.

**Pull request merged**